finesse/gui/interferometer_monitor.py

```diff
@@ -129,14 +89,31 @@ def _get_prop_lineedit(self, prop: EM27Property) -> QLineEdit:
 
     def _display_props(self) -> None:
         """Creates and populates the widgets to view the EM27 properties."""
+        # TODO: remove/hide row from table if it is no longer in _data_table
         for prop in self._data_table:
             lineedit = self._get_prop_lineedit(prop)
             lineedit.setText(prop.val_str())
 
-    def poll_server(self) -> None:
+    def _set_data_table(self, data: list[EM27Property]):
+        """Receive the data table from the server.
+
+        Args:
+            data: the data received from the server
+        """
+        self._data_table = data
```

Comment:
Maybe mention in the docstring that this function is needed because it is use via `pubsub`, otherwise, I would make the same comment as before - why you need a method that just set the data.